src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -8,22 +8,28 @@ import org.apache.log4j.Logger
 import org.apache.mesos.Protos.Offer
 
 import scala.collection.JavaConverters._
+import scala.collection.immutable.Seq
 
 object ResourceMatcher {
   type Role = String
 
   private[this] val log = Logger.getLogger(getClass)
 
-  case class ResourceMatch(cpuRole: Role, memRole: Role, diskRole: Role, ports: RangesResource)
+  case class ResourceMatch(cpuRole: Role, memRole: Role, diskRole: Role, ports: Seq[RangesResource])
+
+  def matchResources(offer: Offer, app: AppDefinition, runningTasks: => Set[MarathonTask],
+                     acceptedResourceRoles: Set[String] = Set("*")): Option[ResourceMatch] = {
 
-  def matchResources(offer: Offer, app: AppDefinition, runningTasks: => Set[MarathonTask]): Option[ResourceMatch] = {
     val groupedResources = offer.getResourcesList.asScala.groupBy(_.getName)
 
-    def findScalarResourceRole(tpe: String, value: Double): Option[Role] = groupedResources.get(tpe).flatMap {
-      _.find { resource =>
-        resource.getScalar.getValue >= value
-      }.map(_.getRole)
-    }
+    def findScalarResourceRole(tpe: String, value: Double): Option[Role] =
+      groupedResources.get(tpe).flatMap {
+        _
+          .filter(resource => acceptedResourceRoles(resource.getRole))
+          .find { resource =>
+            resource.getScalar.getValue >= value
+          }.map(_.getRole)
+      }
 
     val cpuRoleOpt = findScalarResourceRole(Resource.CPUS, app.cpus)
```

Comment:
these are very, very cheap, this has nothing to do with the PR but if I can make you happy that easily ;)